llvm/lib/CodeGen/AsmPrinter/AsmPrinterInlineAsm.cpp

```diff
@@ -400,7 +400,6 @@ void AsmPrinter::emitInlineAsm(const MachineInstr *MI) const {
         "Reserved registers on the clobber list may not be "
         "preserved across the asm statement, and clobbering them may "
         "lead to undefined behaviour.";
-    assert(MMI && "MMI can not be nullptr!");
```

Comment:
Note also that we MMI dereference expresses non-nullness intention, so the assert is not needed.